contracts/rateproducer/rateproducer.cpp

```diff
@@ -4,257 +4,470 @@
-                sum += bp_rate->community;
+            if (community){
+                sum += community;
                 if(row.community){
-                    bp_rate->community = (bp_rate->community + row.community)/2;
+                    community = (community + row.community)/2;
                 }
-                row.community = bp_rate->community;
+                row.community = community;
                 counter++;
             }
 
             if(counter){
-                row.proxy_voters_cntr++;
+                row.ratings_cntr++;
                 row.average =( (sum/counter) + row.average ) /2;
-                row.updated_at = now;
             }
          });
       }
     }
-
-    // for dev only
-    ACTION erase(string table) {
-      // table = bps or stats
-      //only contract owner can erase table
-      require_auth(_self);
-
-      if(!table.compare("bps")){
-      	producers_table bps(_self, _self.value);
-      	auto itr = bps.begin();
-      	while ( itr != bps.end()) {
-      	    itr = bps.erase(itr);
-      	}
+    
+    void update_bp_stats (name * user,
+                       name * bp_name,
+                       float * transparency,
+                       float * infrastructure,
+                       float * trustiness,
+                       float * community,
+                       float * development,
+                       uint32_t * ratings_cntr,
+                       float * average
+                       ){
+      producers_stats_table bps_stats(_self, _self.value);
+      auto itr = bps_stats.find(bp_name->value);
+      if(itr != bps_stats.end()){
+        bps_stats.modify(itr,_self, [&]( auto& row ) {
+            row.transparency = *transparency;
+            row.infrastructure = *infrastructure;
+            row.trustiness = *trustiness;
+            row.development = *development;
+            row.community = *community;      
+            row.ratings_cntr= *ratings_cntr;
+            row.average = *average;
+         });
+          
       }
+    }
+    
+    void calculate_bp_stats (name * bp_name,
+                       float * transparency,
+                       float * infrastructure,
+                       float * trustiness,
+                       float * community,
+                       float * development,
+                       uint32_t * ratings_cntr,
+                       float * average
+                       ){
+        
+        uint32_t category_counter = 0;
+        
+        float transparency_total  = 0;
+        float infrastructure_total = 0;
+        float trustiness_total = 0;
+        float community_total = 0;
+        float development_total = 0;
+        
+        uint32_t transparency_cntr = 0;
+        uint32_t infrastructure_cntr = 0;
+        uint32_t trustiness_cntr = 0;
+        uint32_t community_cntr = 0;
+        uint32_t development_cntr = 0;
+        
+        producers_table bps(_self, _self.value);
```

Comment:
@kriskoin  I beleive it would be better to name this table `ratings` as it is storing user ratings more than information on producers.

what do you think?

Reply:
@xavier506  100% agree , i'll change it

Reply:
@xavier506  done!!